### PR TITLE
Fix: Jarmen Kell recoil animation aborted after kill shot

### PR DESCRIPTION
**Pull request by commy2**
This PR fixes:

- Jarmen Kell may display permanent muzzle flash in transition animations such as while paradropping (map.ini)
- Jarmen Kell has no visible muzzle fire when sniping infantry (muzzle flash however is shown, no issue when sniping vehicles)
- Jarmen Kell recoil animation is aborted prematurely (unless shooting the ground)
- Recoil animation can be cancelled by pressing S-Key (this does not make Jarmen shoot faster however)

---

Jarmen Kell has a neat recoil animation that you never see, because every shot he makes is a kill, and because everything gotta be bugged, the recoil animation is not shown after a kill.

You can still see it by shooting the ground.

https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/70cabb10-dee5-431f-b711-1f746ccc3d7e

Video above has vGLA Jarmen Kell with fix applied, and 1.04 state Demo Jarmen Kell.

All these changes are entirely cosmetical.

### CODE REVIEW

**Comment by commy2** [comment]
This also fixes the recoil animation being cancelled by pressing the STOP button and/or S key.

File: Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+date: 2023-08-21
+
+title: GLA Jarmen Kell recoil animation aborted after kill shot
```
Comment (xezon) [inline Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml]:
```suggestion
title: Fixes broken kill shot recoil animation of GLA Jarmen Kell
```

File: Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+changes:
+  - fix: GLA Jarmen Kell no longer displays permanent muzzle flash in transition animations such as while paradropping.
+  - fix: GLA Jarmen Kell now has a visible muzzle fire in addition to the muzzle flash when sniping infantry.
```
Comment (xezon) [inline Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml]: Perhaps the muzzle flash fixes should be in a different file or are these 3 closely related?

File: Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  - bug
+  - gla
+  - minor
+  - v1.0
+
+links:
+  - https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2239
+
+authors:
+  - xezon
```
Comment (xezon) [inline Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml]: commy2

Comment (commy2) [reply Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml]: copy paste 🤦

Comment (commy2) [reply Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml]:
```suggestion
  - commy2
```

File: Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml (hunk 2 above)
Comment (commy2) [reply Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml]: 1) Could be placed here: https://github.com/TheSuperHackers/GeneralsGamePatch/pull/938

2) is part of the firing animation itself, just like the recoil. See https://github.com/TheSuperHackers/GeneralsGamePatch/pull/1340 or https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2200 for somewhat similar issues.

Comment (xezon) [reply Patch104pZH/Design/Changes/v1.0/2239_jarmen_kell_firing_animation.yaml:8]: You will find a place

**Review by xezon — APPROVED** [review]

File: Patch104pZH/GameFilesEdited/Data/INI/Object/GC_Chem_GLAUnits.ini [hunk 4]
```diff
@@ -3509,7 +3509,7 @@ Object GC_Chem_GLAInfantryJarmenKell
   ; Patch104p @bugfix commy2 10/08/2023 Fix rough transition when injured.
   ; Patch104p @bugfix commy2 18/08/2023 Play longer cheering animation when healthy. (#2238)
   ; Patch104p @bugfix commy2 21/08/2023 Fix recoil animation not always being played. (#2239)
-  ; Patch104p @bugfix commy2 21/08/2023 Fix muzzle fire not shown when sniping infantry. (#2239)
+  ; Patch104p @bugfix commy2 21/08/2023 Fix muzzle fire not shown when sniping infantry. (#938)
```
Comment (xezon) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/GC_Chem_GLAUnits.ini:3512]: This is not the correct number. It should have stayed at #2239, because pull #2239 added it.

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/GC_Chem_GLAUnits.ini:3512]: ok. Follow up PR?

Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/GC_Chem_GLAUnits.ini:3512]: Yes you can tweak it in a future Pull along the side. Not important.
